OPAL/tac/src/main/scala/org/opalj/tac/cg/android/AndroidICCAnalysis.scala

```diff
@@ -1328,7 +1279,19 @@ class AndroidICCAnalysis(val project: SomeProject) extends FPCFAnalysis {
             clone
         }
     }
+}
 
+object StringFinder {
+    def findString(tacCode: TACode[TACMethodParameter, DUVar[ValueInformation]], defSite: Int): Option[String] = {
+        if (defSite > -1) {
+            val expr = tacCode.stmts(defSite).asAssignment.expr
+            if (expr.isStringConst) {
+                Some(expr.asStringConst.value)
+            } else if (expr.isGetStatic) {
+                Some(expr.asGetStatic.name)
```

Comment:
You are right the name does not always equal the value. Therfore this will be changed to the fields value. Is the "findField" method of classfile the right way to access it?

Reply:
No, there just is no proper way to access values in fields. Analyzing this is far from trivial. We do have a points-to analysis that can be used to find some cases with constant values stored in fields, but in general, it is not possible to get all the possible values stored in a field.
So you probably have to see how important this is for your analysis and how you can approximate it.